### PR TITLE
feat: Add function to validate an identifier

### LINKED ISSUE #350: [FEATURE] Function to check for validity of a prerelease identifier
# What / Why
Problem described in #349: An 'invalid' identifier will lead to an invalid `semver`.
Now, there is no way to check beforehand whether an identifier is valid, except doing this:
```
const isValidIdentifier(identifier) => !!valid(inc(parse('0.0.1'), 'prerelease', identifier));
```
This is a request for a more straight-forward function, e.g. `identifier()`, which returns `null` (or `false`) if the identifier is invalid.

## References
  * Related to #349

### PR DESCRIPTION
**Pull request by boxcee**
## References
Closes #350

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -398,6 +399,7 @@ strings that they parse.
   or comparators intersect.
 * `parse(v)`: Attempt to parse a string as a semantic version, returning either
   a `SemVer` object or `null`.
+* `identifier(v)`: Return a valid prerelease identifier, or null if it's not valid.
```
Comment (ljharb) [inline README.md:402]:
```suggestion
* `preIdentifier(v)`: Return a valid prerelease identifier, or null if it's not valid.
```
let's make sure this is clearly a *prerelease* identifier

File: index.js [hunk 2]
```diff
@@ -45,4 +45,5 @@ module.exports = {
   intersects: require('./ranges/intersects'),
   simplifyRange: require('./ranges/simplify'),
   subset: require('./ranges/subset'),
+  identifier: require('./functions/identifier'),
```
Comment (ljharb) [inline index.js:48]:
```suggestion
  preIdentifier: require('./functions/preIdentifier'),
```

**Review by ljharb — COMMENTED** [review]
It'd be awesome to see this rebased so it can be landed :-)

**Comment by wraithgar** [comment]
https://github.com/npm/node-semver/pull/534

**Pull request closed without merging**